### ISSUE TITLE
Enable tests and examples by default

### ISSUE THREAD

**Issue by mhoemmen** [kokkos/stdBLAS#55]
Tests and examples are currently "Off" by default.  This means that building the library doesn't necessarily test anything, or even build the tests or examples.

This might depend on #2.  We'll at least need to change the automatic tests to get gtest as well as mdspan.

**Comment by dalg24** [comment]
Why do you think either should be ON by default?

**Comment by mhoemmen** [comment]
@dalg24 Good question -- some users might not want them.  My personal feeling is that tests and examples should always build by default, as a sanity check for users.  On the other hand, building them requires gtest.  @amklinv-nnl did a fantastic job convincing github to fetch and build all those things, so github's automatic tests already build and run the tests.  I'll be happy to close this as WONTFIX if y'all think it's not a problem.

**Comment by dalg24** [comment]
I lean towards WONTFIX
